src/utils/validation.ts

```diff
@@ -134,6 +134,38 @@ export function validateUint8ArrayLen(name: string, value: unknown, valueKind: s
   }
 }
 
+/**
+ * Validates the given value using all the given fns and throws if all of them fail.
+ *
+ * @param fns - The functions to validate the union with.
+ * @param name - The name of the value.
+ * @param value - The actual value.
+ * @param valueKind - The kind of value that is being checked (e.g. "parameter", "response field", etc.)
+ * @param expected - The expected union type.
+ * @throws - Will throw if not a valid union of the given functions.
+ */
+export function validateUnion(
+  fns: Array<(name: string, value: unknown, valueKind: string) => void>,
+  name: string,
+  value: unknown,
+  valueKind: string,
+  expected: string
+): void {
+  let success = false;
+  for (const fn of fns) {
+    try {
+      fn(name, value, valueKind);
+      success = true;
+    } catch {
+      // Don't propogate thrown error.
+    }
+  }
+
+  if (!success) {
+    throwValidationError(name, value, valueKind, expected);
+  }
+}
```

Comment:
I find this function confusing. When I read "validate union", what I understand it "this function returns a positive outcome if the value given is valid according to **all validators**" and not "this function returns a positive outcome if the value given is valid according to **at least one validator**".

I agree that this is covered by the docstring but I can also say that I read the docstring and I somehow didn't pick up the last part of it.

I would recommend to rename this to `validateAny`.

Reply:
I named it that because it validates a union type, which can be one of multiple types, but I can see how it can be confusing...!